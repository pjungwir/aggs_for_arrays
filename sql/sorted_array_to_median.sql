SELECT sorted_array_to_median('{}'::float[]);
SELECT sorted_array_to_median('{0,1,1,2,5}'::smallint[]);
SELECT sorted_array_to_median('{0,1,1,2,5}'::integer[]);
SELECT sorted_array_to_median('{0,1,1,2,5}'::bigint[]);
SELECT sorted_array_to_median('{0,1,1,2,5}'::real[]);
SELECT sorted_array_to_median('{0,1,1,2,5}'::float[]);
SELECT sorted_array_to_median('{0,1,1,2,5}'::text[]);
SELECT sorted_array_to_median('{0,1,1,2,2,5}'::smallint[]);
SELECT sorted_array_to_median('{0,1,1,2,2,5}'::integer[]);
SELECT sorted_array_to_median('{0,1,1,2,2,5}'::bigint[]);
SELECT sorted_array_to_median('{0,1,1,2,2,5}'::real[]);
SELECT sorted_array_to_median('{0,1,1,2,2,5}'::float[]);
