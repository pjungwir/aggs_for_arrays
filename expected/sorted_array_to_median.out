SELECT sorted_array_to_median('{}'::float[]);
 sorted_array_to_median 
------------------------
                       
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,5}'::smallint[]);
 sorted_array_to_median 
------------------------
                      1
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,5}'::integer[]);
 sorted_array_to_median 
------------------------
                      1
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,5}'::bigint[]);
 sorted_array_to_median 
------------------------
                      1
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,5}'::real[]);
 sorted_array_to_median 
------------------------
                      1
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,5}'::float[]);
 sorted_array_to_median 
------------------------
                      1
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,5}'::text[]);
ERROR:  Median subject must be SMALLINT, INTEGER, BIGINT, REAL, or DOUBLE PRECISION values
SELECT sorted_array_to_median('{0,1,1,2,2,5}'::smallint[]);
 sorted_array_to_median 
------------------------
                    1.5
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,2,5}'::integer[]);
 sorted_array_to_median 
------------------------
                    1.5
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,2,5}'::bigint[]);
 sorted_array_to_median 
------------------------
                    1.5
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,2,5}'::real[]);
 sorted_array_to_median 
------------------------
                    1.5
(1 row)

SELECT sorted_array_to_median('{0,1,1,2,2,5}'::float[]);
 sorted_array_to_median 
------------------------
                    1.5
(1 row)

